athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -147,38 +151,54 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
                 GeneratedRowWriter rowWriter = createFieldExtractors(recordsRequest);
 
                 // Create a new search-source injected with the projection, predicate, and the pagination batch size.
-                SearchSourceBuilder searchSource = new SearchSourceBuilder().size(QUERY_BATCH_SIZE)
+                SearchSourceBuilder searchSource = new SearchSourceBuilder()
+                        .size(QUERY_BATCH_SIZE)
                         .timeout(new TimeValue(queryTimeout, TimeUnit.SECONDS))
                         .fetchSource(ElasticsearchQueryUtils.getProjection(recordsRequest.getSchema()))
                         .query(ElasticsearchQueryUtils.getQuery(recordsRequest.getConstraints().getSummary()));
+
                 // Create a new search-request for the specified index.
-                SearchRequest searchRequest = new SearchRequest(index).preference(shard);
-                int hitsNum;
-                int currPosition = 0;
-                do {
-                    // Process the search request injecting the search-source, and setting the from position
-                    // used for pagination of results.
-                    SearchResponse searchResponse = client
-                            .getDocuments(searchRequest.source(searchSource.from(currPosition)));
-
-                    // Throw on query timeout.
+                SearchRequest searchRequest = new SearchRequest(index)
+                        .preference(shard)
+                        .scroll(TimeValue.timeValueSeconds(60))
```

Comment:
This means the ElasticSearch search request will stop maintaining context of the state of our search after 60 seconds (docs say `The scroll parameter indicates how long Elasticsearch should retain the [search context](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#scroll-search-context) for the request.` [link](https://www.elastic.co/guide/en/elasticsearch/reference/current/scroll-api.html#scroll-api)). What is the expected behavior if a query to ES takes more than a minute? Have we tested this with very large ES indices to verify we don't approach this limit in our handlers?